### DISCUSSION TITLE
Notes from this weekend

### DISCUSSION THREAD

**Discussion by coverbeek** [frc5567/2023-Robot#178] [Team Posts]
It was good to see most of you this weekend. Great discussion today and some great ideas for our bot this year.

I'm going to be very honest when I say this will likely be the most complex bot Team 5567 has attempted and the coaches and mentors are excited to try to help you all make it happen. This year, at least initially, Danny is going to take a more active build role in an attempt to ensure that wiring and electrical are considered as part of all designs. I will be working more closely with incoming students and our experienced coders to help get others up to speed while laying the groundwork for our 2020 robot code and getting the scouting application updated to the needs of the coaches/scout team.

I've created the 2020-Robot repository and will do the same for scouting tomorrow.
I made a bunch of notes about things that we will need to research or be ready to handle.

- Need to research LimeLight and it's APIs and capabilities. We will need to use all of it's capabilities with respect to calculating distance to target, angle to target, and potentially calculating changes in position needed to hit target based on tuned/tested values. [http://docs.limelightvision.io/en/latest/](http://docs.limelightvision.io/en/latest/)
- Drivetrain is likely to be similar to the last couple years, with some improvements. Controls are going to depend on our drivers' preferences. We should get a rudimentary implementation in place that can give the drivers a couple of approaches to try and vote on. 
- An active pick-up mechanism is going to require code to deploy/retract, to enable and disable retrieval.
- Discussion about hopper/holding/feeder mechanism ranged from nearly passive to completely active and indexed. We should be prepared to code controls for intake and expel (motors moving power cells within mechanism) and possibly controlled feed to shooter mech.
- In order to shoot at the high target, our shooter is at the very least going to have some kind of high-speed flywheel for launch, possibly also having a secondary motor/pneumatic for powered-feed to the launcher
- For the climber we are going to minimally need controlled extension and retraction of the climbing mechanism. There is a possible need also for ability to adjust center of mass to help balance. More discussion of the actual design will dictate our efforts here.
- For the control panel, we should be able to handle the 3 revolutions relatively safely with encoders, but for the color sensing we need to research what we can do. We *should* be able to use either a color sensor or the PixyCam.
- For Auton, we're going to need to move based on distance (encoders), align and shoot based on vision, and potentially move and retrieve additional power cells for shooting. I would like to get a skeleton auton put together as early as possible with placeholders for the scoring mechanisms.

For now, these are just my notes based on preliminary discussions about what we could do to put in place a skeleton robot code implementation. Ideally we would create classes for the major conceptual objects and try to pattern the behaviors as methods on those classes. We probably won't get that far this week, but in order for prototyping to happen quickly we need to be prepared to put code together to allow rudimentary control of prototypes. As requirements become clearer we will need to move quickly, so I'm trying to make sure you start thinking about how we might control the various systems proposed today. Some of those ideas are going to be incorporated into the designs that eventually become our working robot.

Welcome back to the most wonderful time of year! 
-Chris (and Danny by proxy)

**Comment by MikasaSucasa** [comment]
Thanks Chris! I only write long notes at 3am or during work ;P. I will be available for questions and discussions 24/7 and I will be stepping in/out for Chris's absences.
Be sure to re-download all of the libraries and VScode!

**Comment by coverbeek** [comment]
The documentation for the control system has been completely reworked. Very nice.
[https://docs.wpilib.org/en/latest/docs/getting-started/getting-started-frc-control-system/intro.html](https://docs.wpilib.org/en/latest/docs/getting-started/getting-started-frc-control-system/intro.html)

**Comment by coverbeek** [comment]
FAMNM is doing Robot in 3 Days. Good stuff here about how they approach the process...
[https://www.youtube.com/watch?v=_RTVrUvHmI4](https://www.youtube.com/watch?v=_RTVrUvHmI4)
[https://www.youtube.com/watch?v=T5QxBuHllm0](https://www.youtube.com/watch?v=T5QxBuHllm0)
[https://www.youtube.com/watch?v=Nis5KBcncSw](https://www.youtube.com/watch?v=Nis5KBcncSw)

**Comment by MikasaSucasa** [comment]
Everyone is busy at work and school I see ;P
